bpf/bpf_sock.c

```diff
@@ -89,15 +138,106 @@ int sock4_xlate(struct bpf_sock_addr *ctx)
 		if (!backend)
 			return CONNECT_PROCEED;
 
+		if (ctx->protocol != IPPROTO_TCP &&
+		    sock4_update_revnat(ctx, backend, &key,
```

Comment:
Maybe it makes sense to log if `map_update_elem()` has failed in `sock4_update_revnat()`. Can be done in a separate PR.

Reply:
There is currently no perf event output support in the kernel for this type, but we can add it to the kernel. I'll add a separate metric map entry, along with the net-next box update in a next pr.